### PR TITLE
fix: do pass tipsets_included to TipsetRangeSyncer

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:
- original code looks buggy, it's a compilation error if we use `hashbrown::HashMap` here 

## Reference issue to close (if applicable)

Closes 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
Could you please double-check that this works fine on clean mainnet?

File: blockchain/chain_sync/src/tipset_syncer.rs [hunk 1]
```diff
@@ -670,11 +670,11 @@ where
         ));
 
         let mut tipsets_included = HashSet::new();
-        tipsets_included.insert(proposed_head.key());
+        tipsets_included.insert(proposed_head.key().clone());
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply blockchain/chain_sync/src/tipset_syncer.rs]: Fixed. that's much better!

**Review by lemmih — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
@hanabi1224 I've been running this branch for a while, and I confirm it works fine. Thanks!

**Comment by hanabi1224** [comment]
@LesnyRumcajs thanks!

**Pull request merged**